templates/funcs.go

```diff
@@ -221,11 +232,29 @@ func render(f pgs.Field) string {
 
 	var s string
 
+	var obsType zap.ObfuscationType
+	if _, err := f.Extension(zap.E_ObfuscationType, &obsType); err != nil {
+		panic(fmt.Sprintf("error getting obfuscation_type for field %s: %s", f.Name(), err))
+	}
+
+	if obsType == zap.ObfuscationType_HIDE {
+		return ""
+	}
+
 	// repeated
 	if t.IsRepeated() {
 
-		if t.Element().IsEnum() || (t.Element().IsEmbed() && t.Element().Embed().IsWellKnown()) {
-			d := newArrayData("Stringers", getter(n, t), name(f))
+		if obsType == zap.ObfuscationType_STARS {
+			d := newArrayData("StringArray", getter(n, t), name(f), obsType)
```

Comment:
If it's possible (Im not 100% sure), I would offload this choice of changing the slice type to StringArray for STARS obfuscation to `newArrayData`, I think it's more its role and should prevent you from having to update this part of the code